src/Controller/UserController.php

```diff
@@ -121,19 +132,53 @@ function postDeleteAction(Application $app, $userName) {
 
     function calendarAction(Application $app, $userName) {
         $calendars = $app['sabredav.backend.caldav']->getCalendarsForUser('principals/' . $userName);
+        $calendarsData = [];
 
+        foreach ($calendars as $calendar) {
+            $calendarId = $calendar['id'];
+            $calendar['eventCount'] = count($app['sabredav.backend.caldav']->getCalendarObjects($calendarId));
+            $calendarsData[] = $calendar;
+        }
+        #return json_encode($calendarsData);
         return $app['twig']->render('admin/user/calendars.html', [
             'username'  => $userName,
-            'calendars' => $calendars,
+            'calendars' => $calendarsData,
         ]);
     }
 
     function addressbookAction(Application $app, $userName) {
         $addressbooks = $app['sabredav.backend.carddav']->getAddressBooksForUser('principals/' . $userName);
+        $addressbooksData = [];
 
+        foreach ($addressbooks as $addressbook) {
+            $addressbookId = $addressbook['id'];
+            $addressbook['cardCount'] = count($app['sabredav.backend.carddav']->getCards($addressbookId));
+            $addressbooksData[] = $addressbook;
+        }
+        #return json_encode($addressbooksData);
         return $app['twig']->render('admin/user/addressbooks.html', [
             'username'     => $userName,
-            'addressbooks' => $addressbooks,
+            'addressbooks' => $addressbooksData,
         ]);
     }
+
+    function deleteAddressbookAction(Application $app, $userName, $addressbookId) {
```

Comment:
I worry about using the id as the primary identifer. I'm gonna merge this and change it to the uri. The `uri` is more predictably a string.